docs/components/styling.md

```diff
@@ -233,6 +233,66 @@ By specifying "inner-text" as the value of the `exportparts` attribute, elements
 <outer-component />
 ```
 
+## Style Modes
+
+Component Style Modes enable you to create versatile designs for your components by utilizing different styling configurations. This is achieved by assigning the styleUrls property of a component to a collection of style mode names, each linked to their respective CSS files.
+
+### Example: Styling a Button Component
+
+Consider a basic button component that supports both iOS and Material Design aesthetics:
+
+```tsx
+@Component({
+  tag: 'simple-button',
+  styleUrls: {
+    md: './simple-button.md.css', // styles for Material Design
+    ios: './simple-button.ios.css' // styles for iOS
+  },
+})
+export class SimpleButton {
+  // ...
+}
+```
+
+To dictate the style mode (Material Design or iOS) in which the button should be rendered, you must initialize the desired mode before any component rendering occurs. This can be done as follows:
+
+```ts
+import { setMode } from '@stencil/core';
+setMode(() => 'ios'); // Setting iOS as the default mode for all components
+```
+
+The `setMode` function processes all elements, enabling the assignment of modes individually based on specific element attributes. For instance, by assigning the `colorMode` attribute to a component:
+
+```html
+<simple-button colorMode="ios"></simple-button>
+```
+
+You can conditionally set the style mode based on the `colorMode` property:
+
+```ts
+import { setMode } from '@stencil/core';
+
+const defaultMode = 'md'; // Default to Material Design
+setMode((el) => el.getAttribute('colorMode') || defaultMode);
+```
+
+Style modes can only be set once at the beginning of the component lifecycle.
+
+### Important Considerations
+
+- __Initialization:__ Style modes must be defined before the components gets rendered into the DOM and can't be changed afterwards. Updating the style mode requires manually re-rendering of all components.
+- __Usage Requirement:__ A style mode must be set to ensure the component loads with styles. Without specifying a style mode, the component will not apply any styles.
```

Comment:
Can we add some additional information regarding what happens if:
- We mistype the mode (e.g. `setMode('oos')`)
- A component doesn't have any style that matches the arg passed to `setMode`, but other components do?

Reply:
Good catch 👌 

> We mistype the mode (e.g. setMode('oos'))

I will just point out that a user should ensure that set modes should get validated otherwise components remain unstyled.

> A component doesn't have any style that matches the arg passed to setMode, but other components do?

A style mode assigned to a component that doesn't defines that style mode will remain unstyled.

Reply:
I wonder if there is a way to get all defined style modes 🤔 if components define different style modes there might use cases where one would like to know which are supported by a component.

Reply:
There isn't for users today, but I don't think that'd be too big of a lift - we have that style [mode resolution list](https://github.com/ionic-team/stencil/blob/80ef14361613706d38ad6799a03d3e5f1e1e6cbe/src/runtime/mode.ts#L9) _and_ this should be pretty easy for us to derive for specific components at compile time for more details (e.g. you wanted to see if any components are missing a mode of some particular name).

The Framework team/Mike H have asked us about a typed version of `getMode`/`setMode` in the past as well - one where you can specify _somewhere_ the allowable types/strings on those methods to prevent typos in the first place. I don't think we have something in the backlog, but could be something worth asking in our internal ask_framework channel.

Reply:
Definitely worth adding IMHO. Added the following limitation to https://github.com/ionic-team/stencil-site/pull/1342/commits/a329dd5fd6909686b27b262bf580067f673e9901:
```
- __Input Validation:__ Verify a style mode is supported by a component you are setting it for. Setting an un-supported style mode keeps the component unstyled.
```

Reply:
👍 Can you do me a favor and add a ticket to the backlog for us to look into?

Reply:
Ingested into backlog as `STENCIL-1124`